internal/encoder.go

```diff
@@ -378,7 +378,25 @@ func CastValue(t types.Type, v Value) (Value, error) {
 			return nil, err
 		}
 		return FloatValue(f64), nil
-	case types.STRING, types.ENUM:
+	case types.STRING:
+		switch v.(type) {
+		// If this is coming from a date/time, the format is slightly different
+		// than when just writing the value out as a string.
+		case DateValue:
```

Comment:
Fix the `ToString` method side that is implemented in each `DateValue` etc.
FYI: https://github.com/goccy/go-zetasqlite/blob/main/internal/value.go#L1574-L1576

Reply:
@colincadams Do you have time to this fix ?

Reply:
Hi @goccy, I believe I've implemented the requested changes.

`ToString()` now returns the string that we use when casting
`ToApiString()` now returns the string representation of the `Value` that is used in the BigQuery API.